osu.Game/Online/Multiplayer/MultiplayerClient.cs

```diff
@@ -189,5 +179,65 @@ public override Task StartMatch()
 
             return connection.InvokeAsync(nameof(IMultiplayerServer.StartMatch));
         }
+
+        private async Task disconnect(bool takeLock)
+        {
+            cancelExistingConnect();
+
+            if (takeLock)
+                await connectionLock.WaitAsync();
+
+            try
+            {
+                if (connection != null)
+                    await connection.StopAsync();
```

Comment:
Given that each `Connect()` spawns a new connection now, this should probably be

```suggestion
                    await connection.DisposeAsync();
```